android/app-project-newm/src/main/java/io/projectnewm/screens/login/password/PasswordState.kt

```diff
@@ -0,0 +1,34 @@
+package io.projectnewm.screens.login.password
+
+import io.projectnewm.screens.login.TextFieldState
+
+class PasswordState : TextFieldState(validator = ::isPasswordValid, errorFor = ::passwordValidationError)
+
+class ConfirmPasswordState(private val passwordState: PasswordState) : TextFieldState() {
+    override val isValid
+        get() = passwordAndConfirmationValid(passwordState.text, text)
+
+    override fun getError(): String? {
+        return if (showErrors()) {
+            passwordConfirmationError()
+        } else {
+            null
+        }
+    }
+}
+
+private fun passwordAndConfirmationValid(password: String, confirmedPassword: String): Boolean {
+    return isPasswordValid(password) && password == confirmedPassword
+}
+
+private fun isPasswordValid(password: String): Boolean {
```

Comment:
Use shared password validator.  Right now it just checks if the field is blank so you could add these checks there as well